### ISSUE TITLE
Add information about State sync parts into debug page

### ISSUE THREAD

**Issue by mm-near** [near/nearcore#7991]
One of the parts of node syncing process is 'State sync' (see more info in: https://near.github.io/nearcore/architecture/how/sync.html#step-2-state-sync-normal-node)

The goal of this part is to download the current state (whole TRIE) from other participants. It is done by requesting 'parts' (each around 1MB). We already have debug information on the node who is syncing (requesting) these parts.

What we're missing, is debug information on the node from which parts are requested.

There are 2 crucial information that we'd like to have:
* which parts were already requested
* how long did it take to create those parts

The main code creating these parts is here:
https://github.com/near/nearcore/blob/869a0110d1e10ee7588875778a01e37c5e85c6b5/chain/chain/src/chain.rs#L2820

As you can see, we first check if the part wasn't created in the past (by looking into DBCol::StateParts) - and if it wasn't - we create it (in obtain_state_part) and save to DB.

# What should be done:

We should create a new debug API entry point and new request type:
https://github.com/near/nearcore/blob/869a0110d1e10ee7588875778a01e37c5e85c6b5/chain/jsonrpc/src/lib.rs#L789

https://github.com/near/nearcore/blob/869a0110d1e10ee7588875778a01e37c5e85c6b5/chain/client/src/debug.rs#L172

And when such request is received - return which parts we already have in storage.

Optionally - we could also persist (in memory only) - the time it took us to create such part (time spent in 'obtain_state_part' method) - and return it in this json API too.

Yet another cool thing - would be to add this information to one of our debug pages (for example sync.html):
https://github.com/near/nearcore/blob/869a0110d1e10ee7588875778a01e37c5e85c6b5/chain/jsonrpc/res/sync.html#L272

**Comment by askoa** [comment]
I am going to make an attempt. I have some questions.

1. "which parts were already requested" --> I am assuming in memory cache should be used to store the parts that were already requested. 
    Reason for the above assumption: I am not sure if `Chainstore->store` can be queried for all items in column `DBCol::StateParts`. Even if there is a way to query it, I am not sure how to get `shard_id, part_id` from the `key`.
2. Is in memory cache same as the `CellLruCache` in `ChainStore`? If they are _not_ the same, can you provide some samples of in memory cache usage in the current code? 
3. I am assuming the response will have shard_id, part_id and the time taken to create the part_id. Should the response have other fields? for e.g. the `key` for `DBCol::StateParts` uses `CryptoHash`. Should `CryptoHash` be included in the response?
4. What's the max number of items expected in the response? I see the existing LRU cache is created with `CHUNK_CACHE_SIZE = 1024` or `CACHE_SIZE = 100`. If the response is going to come from LRU cache, just wanted to make sure the items fits within the bounds of current cache size.

**Comment by mm-near** [comment]
@askoa - Great, Thank you !

1. Sure we can start with in-memory (should be easier & faster). 'store' has iter_prefix function, that would do what you want, but AFAIK it also reads the content, which might be quite slow for this use case.
StatePartKey is 'borsh_serialized' into the string (you can read more about borsh here:
https://near.github.io/nearcore/architecture/how/serialization.html#borsh) - but as StateKey is a simple tuple - it is basically a concatenation: of CryptoHash, ShardId and number.
Anyways - as I've mentioned above - we can start with just-in-memory for it.

2. yep, CellLRUCache should work well for this scenario
3. CryptoHash is quite important: it is the EpochId from which the state is requested. So it should be the 'main' index in the response. The way to think about it is like this: Some new node is trying to get the State of a given Shard at a given moment in time (Epoch boundary) - as states are large, it requests it in parts.
So basically the response should be something like: 
["B1KygeFajWTRUSEgCSH3Csw5Vh9Fbzu7JLx3WHL8HkCo" : [ "0" : [ "13": 200ms, "15: 400ms ] ]]
Where the highest index is the epoch id (a.k.a CryptoHash) - then we look into shard (0) and then part (13) 

4. At any moment, users can request State only for two possible hashes (basically current and previous epochs). We currently have 4 shards, and each part is around 1MB in size. Our Shards in testnet are around 50GB.
So we'll need 50'000 * 4 * 2 == 400'000 at minimum (to keep the information about possibly all the parts from all shards from the last 2 epochs). So I'd make it 1'000'000.

**Comment by askoa** [comment]
@mm-near Thanks for the response. Is the request suppose to have any parameters like CryptoHash? Or the response should return all the entries in the in-memory LRU cache.
